File: LeetCode/LeetCode Practice Problems/257- Binary Tree Paths.cpp
class Solution {
public:
    vector<string> a;
    void find(TreeNode* root,string s) {
        string t = to_string(root->val);
        s = (s.length() > 0)?(s+"->"):"";
        if(root->left == NULL && root->right == NULL) {
            a.push_back(s+t);
        
            return;
        }
        
        if(root->left != NULL) {
            find(root->left , s+to_string(root->val));
        }
        if(root->right != NULL)
            find(root->right , s+to_string(root->val));
    }
    
    vector<string> binaryTreePaths(TreeNode* root) {
        find(root,"");
        return a;
    }
};

File: LeetCode/LeetCode Practice Problems/257-Binary Tree Paths.java
class Solution {
    
    public void findPaths(TreeNode root,List<String> ans,String cur) {
        if(root == null) return;
        String path = cur;
        if(cur.length() > 0) {
            path = path + "->";
        }
        path = path + Integer.toString(root.val);
        
        if(root.left == null && root.right == null) {
            ans.add(path);
            return;
        }
        findPaths(root.left , ans , path);
        findPaths(root.right , ans , path);
    }
    
    public List<String> binaryTreePaths(TreeNode root) {
        List<String> ans = new ArrayList();
        findPaths(root,ans,"");
        return ans;
    }
}
